### PR TITLE
Implement bare minimum modal component

### LINKED ISSUE #8: [Project - Onboarding]: Implement Modal component (without styling)
# Implement React Component: Modal

## Objective:
The objective of this issue is to create a new React component named `Modal`. This component will be used to display different kinds of modals for the onboarding flow.

### 1. Properties:

| Property  | Type | Description | Required |
| --------- | ---- | ----------- | -------- |
| children  | React.ReactNode | Children elements | Yes |
| onClose   | () => void | Called to close the modal | No |

### 2. Behaviors:

- Overlay and centered: The modal should be centered
- Darken the background: The modal should be on top of a background, which darkens the content below (use the opacity of 30% and color black for now.
- The modals visibility is controlled outside. The parent component decides whether the modal should be displayed or not.
- Use `React.Portal` to display the modal at `document.body` to guarantee a proper overlay of the underlying content

### 3. Design (DRAFT):
image
image

## Acceptance Criteria:

- The component is implemented according to the requirements described above, or an existing component is taken, which fulfills the requirements.
- The component passes all defined test cases and has Stories displaying its states
- The component linked to a figma component in Storybook - speak to designers!
- The component should be responsive and display correctly on browser windows (min-width 1024px)
- You should not implement any complex CSS styles! The design is still work in progress. The component will be finalized design-wise in a separate ticket. Make sure, that the `properties` and the `behaviors` are implemented accordingly.

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/addon-onboarding/issues/8

I have created a basic Modal component based on radix UI. The styling and polishing will be done in a separate task.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.9--canary.18.0c6aa80.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-onboarding@0.0.9--canary.18.0c6aa80.0
  # or 
  yarn add @storybook/addon-onboarding@0.0.9--canary.18.0c6aa80.0
  ```
</details>

### CODE REVIEW

File: src/components/Modal/Modal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {({ Title, Description, Close }) => (
+      <>
+        Hello world
+        <Title>Modal Title</Title>
+        <Description>Modal Description</Description>
+        <Close>Close</Close>
+      </>
+    )}
+  </Modal>
+);
```
Comment (yannbf) [inline src/components/Modal/Modal.stories.tsx]: Shall we make this show an example of a controlled modal + use the play function to make sure things work as intended:

```suggestion
import { Meta, StoryObj } from "@storybook/react";
import { Modal } from "./Modal";
import React, { useState } from "react";
import { userEvent, within } from "@storybook/testing-library";

const meta: Meta<typeof Modal> = {
  component: Modal,
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => {
    const [isOpen, setOpen] = useState(false)

    return (
      <>
        <Modal width="740px" open={isOpen}>
          {({ Title, Description, Close }) => (
            <>
              Hello world
              <Title>Modal Title</Title>
              <Description>Modal Description</Description>
              <Close onClick={() => setOpen(false)}>Close</Close>
            </>
          )}
        </Modal>
        <button onClick={() => setOpen(true)}>Open modal</button>
      </>
    )
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByText("Open modal");
    await userEvent.click(button);
  }
}
```

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  extends Omit<React.ComponentProps<typeof Dialog.Root>, "children"> {
+  width?: string;
+  children: (props: {
+    Title: typeof StyledTitle;
+    Description: typeof StyledDescription;
+    Close: typeof StyledClose;
+  }) => React.ReactNode;
+}
+
+export function Modal({ children, width, ...rootProps }: ModalProps) {
```
Comment (yannbf) [inline src/components/Modal/Modal.tsx]: Should the modal deal with key events like <kbd>escape</kbd>? Maybe `@radix-ui/react-dialog` already has an option to enable that?

File: src/components/Modal/Modal.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { css, keyframes, styled } from "@storybook/theming";
```
Comment (yannbf) [inline src/components/Modal/Modal.styled.tsx]:
```suggestion
import { css, styled } from "@storybook/theming";
```

File: src/components/Modal/Modal.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  Overlay,
+  Content,
+  Title,
+  Description,
+  Close,
+} from "@radix-ui/react-dialog";
+import React from "react";
+
+export const StyledOverlay = styled(Overlay)`
+  background-color: rgba(0, 0, 0, 0.25);
```
Comment (yannbf) [inline src/components/Modal/Modal.styled.tsx:12]: There is a slight flicker once the modal is opened, do you think it'd make sense to add a transition?

**Review by yannbf — APPROVED** [review]
LGTM, with a few minor comments

File: src/components/Modal/Modal.styled.tsx (hunk 4 above)
Comment (valentinpalkovic) [reply src/components/Modal/Modal.styled.tsx:12]: Let's take another look at the styling/polishing Milestone

**Pull request merged**